src/g_array.c

```diff
@@ -475,34 +475,54 @@ void garray_arraydialog(t_garray *x, t_symbol *name, t_floatarg fsize,
 }
 
 /* jsarlo { */
+static int garray_arrayviewlist_setpage(t_symbol*arrayname, t_array*a, int page, int pagesize) {
+
+        /* make szre the requested page is within range */
+    int maxpage = (a->a_n - 1) / pagesize;
+    if(page < 0)
+        page = 0;
+    if(page > maxpage)
+        page = maxpage;
+
+    sys_vgui("::dialog_array::listview_setpage {%s} %d %d %d\n",
+        arrayname->s_name,
+        page, maxpage+1, pagesize);
+    return page;
+}
 void garray_arrayviewlist_new(t_garray *x)
 {
-    int i, yonset=0, elemsize=0;
+    int i, yonset=0, elemsize=0, page=0;
     t_float yval;
     char cmdbuf[200];
     t_array *a = garray_getarray_floatonly(x, &yonset, &elemsize);
 
     if (!a)
     {
-        /* FIXME */
+            /* FIXME */
         pd_error(0, "error in garray_arrayviewlist_new()");
+        return;
     }
     x->x_listviewing = 1;
     sprintf(cmdbuf,
-            "pdtk_array_listview_new %%s %s %d\n",
+            "pdtk_array_listview_new %%s {%s} %d\n",
             x->x_realname->s_name,
             0);
     gfxstub_new(&x->x_gobj.g_pd, x, cmdbuf);
-    for (i = 0; i < ARRAYPAGESIZE && i < a->a_n; i++)
+
+    page = garray_arrayviewlist_setpage(x->x_realname, a, page, ARRAYPAGESIZE);
+
+    sys_vgui("::dialog_array::listview_setdata {%s} %ld",
+        x->x_realname->s_name,
+        (long long)(page * ARRAYPAGESIZE));
+    for (i = page * ARRAYPAGESIZE;
+         (i < (page + 1) * ARRAYPAGESIZE && i < a->a_n);
+         i++)
     {
-        yval = *(t_float *)(a->a_vec +
+        yval = *(t_float *)(a->a_vec + \
```

Comment:
I guess the `(t_float *)` cast is a relict of the 32-bit era. It happens to work, but it would be more correct to cast to the actual type (`t_word`).

Also, what's up with that backslash? :-)

Reply:
Actually, why not simply use `garray_getfloatwords`? For garrays, `elemsize` and `yonset` are not relevant.